zjit/src/hir.rs

```diff
@@ -4694,10 +4848,16 @@ mod tests {
         assert_contains_opcode("test", YARVINSN_newhash);
         assert_snapshot!(hir_string("test"), @r"
         fn test@<compiled>:1:
-        bb0(v0:BasicObject):
-          v5:HashExact = NewHash
+        bb0(v1:BasicObject):
+          EntryPoint interpreter
+          Jump bb2(v1)
+        bb1(v4):
```

Comment:
It's a little odd that the parameters to the JIT entrypoint don't get flow typed. I wonder why that is. They should get `BasicObject`

Reply:
https://github.com/ruby/ruby/pull/14643/commits/a31c8cbdee4c9007b307338a7d9cbae8484481ad